packages/runtime/runtime-definitions/src/id-compressor/idCompressor.ts

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SessionSpaceCompressedId, OpSpaceCompressedId, SessionId, FinalCompressedId, StableId } from "./identifiers";
+import { IdCreationRange, SerializedIdCompressor, SerializedIdCompressorWithNoSession, SerializedIdCompressorWithOngoingSession } from "./persisted-types";
+
+export interface IIdCompressorCore {
```

Comment:
Can you doc these please? I'm not sure what the distinction is between `IIdCompressorCore` and `IIdCompressor`, or why you chose which methods for each.

Reply:
I'll add docs here, but the idea is that `IIdCompressorCore` exposes the `IdCompressor` methods that can "mess up" the internal state of the compressor (namely `finalizeCreationRange` and `takeNextCreationRange`). We need these at the `SharedObjectCore` level to append the `idRange` to the ops and to finalize on incoming ops, but we don't want this to be done by a DDS author. The DDS itself only has access to an `IIdCompressor`, which exposes methods that a DDS author would want access to. Taking a look at it, I think we could probably move the serialization and deserialization methods into `IIdCompressorCore`. I didn't think too long about what methods should go into which interface.

Reply:
I see! Yea, I agree with you. `getAllIdsFromLocalSession` also seems kind of random from the perspective of a DDS author - maybe that should be part of core too.

Reply:
Moved a lot more methods into `IIdCompressorCore` and also added some doc comments to each interface.